### DISCUSSION TITLE
Access entry variables from a template override

### DISCUSSION THREAD

**Discussion by thisisjamessmith** [verbb/formie#2055] [Q&A] [answered]
In the context of a body builder matrix, we typically allow for changing the background colour of each section. When changing the background colour of a Formie block/entry type, how can I access the selected background colour (a Craft dropdown field) from within a Formie template override? Any other way to do this?

**Comment by thisisjamessmith** [comment]
Ah nevermind I think I figured out a solution - I can pass arbitrary variables/objects via the `themeConfig` object, which can then get read by the field templates with `form.themeConfig.whatever`.

**Comment by engram-design** [comment, marked as answer]
You can pass in any cariable you like as [`renderOptions`](https://verbb.io/craft-plugins/formie/docs/theming/render-options#custom-variables).

```twig
{% set renderOptions = {
    myVariable: 'some-value',
} %}

{{ craft.formie.renderForm('contactForm', renderOptions) }}
```

Which are available in any Formie template as `renderOptions.myVariable`.

**Reply by thisisjamessmith** [reply]
Ah even better, thanks Josh :-)
